js/projection/pg-refactory.js

```diff
@@ -0,0 +1,85 @@
+import Promise from 'bluebird';
+import _ from 'underscore';
+import $ from 'jquery';
+import Backbone from 'backbone';
+import TableView from '../../js/table-view.js'
+
+
+function odata (promise, {
+	verb = 'get',
+	url,
+	skip,
+	take,
+	filter,
+	orderby = [],
+	select = [],
+} = {}) {
+
+	var url = _.isFunction(url) ? url() : url;
+	var op = {
+		url: url,
+		$format: 'json',
+		$count: true,
+	};
+
+	if (take) {
+		op.$top = take;
+	}
+
+	if (skip) {
+		op.$skip = skip;
+	}
+
+	if (_.size(orderby)) {
+		var col = _.first(orderby);
+		var key = _.keys(col)[0];
+		var dir = col[key];
+
+		op.$orderby = key + ' ' + (dir > 0 ? 'asc' : 'desc');
+	}
+
+	
+	var promiseData  = new Promise(function (resolve, reject) {
+		$.getJSON(op.url, _.omit(op, 'url'))
+		  .success(resolve)
+		  .fail(function (jqXHR, textStatus, errorThrown) {
+		    reject(new Error(errorThrown));
+		  });
+	});
+
+	return promiseData;
+}
+
+function renderTable (promise, options) {
+	var promiseData = promise.then(function (data) {
+		var columns = [];
+		for (var key in data.value[0]) {
+			columns.push({name: key});
+		}
+
+		var bodyrows = [];
+		for (var i = 0; i < data.value.length; i++) {
+			var obj = {classes: ['body-row']};
+			obj.item = {};
+			for (var key in data.value[i]) {
+				obj.item[key] = {html: data.value[i][key]};
+			}
+			bodyrows.push(obj);
+		}
+
+		const tableView = window.tableView = new TableView({
+			virtualized: true,
+		}).set({
+			columns:columns, 
+			headRows:['column-header-rows'], 
+			bodyRows:bodyrows, 
+		}).render();
+		$(() => tableView.$el.appendTo('body'));
+	});
+	return promiseData;
+}
+
+function map (promise, options) {
```

Comment:
OK~ One more question: should I wrap each function like this: export { function name} ?